src/test/kotlin/com/software/modsen/driverservice/controller/CarControllerIntegrationTest.kt

```diff
@@ -0,0 +1,204 @@
+                jsonPath("$.year") { value(updatedCar.year) }
+            }
+        mockMvc.put("/cars/{id}", DEFAULT_ID)
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = objectMapper.writeValueAsString(defaultCar)
+        }
+    }
+
+    @Test
+    fun `should return list of cars`() {
+        val expectedCarResponseList = listOf(getDefaultCarResponse())
+        mockMvc.get("/cars")
+            .andDo { print() }
+            .andExpect {
+                status { isOk() }
+                content {
+                    RequestPredicates.contentType(MediaType.APPLICATION_JSON)
+                }
+                jsonPath("$.items[0].carId") { value(expectedCarResponseList[0].carId) }
+                jsonPath("$.items[0].model") { value(expectedCarResponseList[0].model) }
+                jsonPath("$.items[0].licensePlate") { value(expectedCarResponseList[0].licensePlate) }
+                jsonPath("$.items[0].color") { value(expectedCarResponseList[0].color) }
+                jsonPath("$.items[0].year") { value(expectedCarResponseList[0].year) }
+            }
+    }
+
+    @Test
+    fun `should return car already exist on create`() {
+        val carRequest = getDefaultCarRequestWithAlreadyExistingLicenseInDB()
+        mockMvc.post("/cars")
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = objectMapper.writeValueAsString(carRequest)
+        }
+            .andDo { print() }
+            .andExpect {
+                status { isConflict() }
+                content {
+                    contentType(MediaType.APPLICATION_JSON)
+                    content {
+                        contentType(MediaType.APPLICATION_JSON)
+                    }
+                    jsonPath("$.message") { value(ExceptionMessages.CAR_ALREADY_EXIST.format(DEFAULT_LICENSE_PLATE)) }
+                }
+            }
+    }
+
+    companion object {
+        private val DEFAULT_ID = 1L
+        private val DEFAULT_LICENSE_PLATE = "1234AB"
+    }
+    private fun getDefaultCarRequest() =
+        CarRequest(
+            model = "Tesla Model S",
+            year = 2022,
+            licensePlate = "1234AB",
+            color = "Black"
+        )
+
+    private fun getDefaultCarResponse() =
+        CarResponse(
+            carId = 1L,
+            model = "Tesla Model S",
+            year = 2022,
+            licensePlate = "1234AB",
+            color = "Black"
+        )
+
+    private fun getDefaultUpdatedCarRequest() =
+        CarRequest(
+            model = "Tesla Model A",
+            year = 2023,
+            licensePlate = "5678BB",
+            color = "White"
+        )
+
+    private fun getDefaultCarRequestWithAlreadyExistingLicenseInDB() =
+        CarRequest(
+            model = "Tesla Model A",
+            year = 2020,
+            licensePlate = "1234AB",
+            color = "White"
+        )
```

Comment:
Better to move under companion object and to make them be vals, not functions

Reply:
Move on vals